lib/r10k/git/shellgit.rb

```diff
@@ -53,26 +74,22 @@ def clean(opts = {})
         end
       end
 
-      def fetch(remote, opts = {})
-        cmd = ["fetch", remote, '+refs/*:refs/*']
-        result = git(cmd, opts)
-
-        if result.success?
-          return true
-        else
-          raise R10K::Git::GitError.new(result.stderr)
+      def clone(remote, local, opts={})
+        if opts[:private_key] || opts[:username]
+          logger.warn("Shellgit provider does not support custom SSH transport options, using default username and/or private key.")
         end
-      end
 
-      def clone(remote, local, opts={})
-        cmd = ["clone", "--mirror"]
+        cmd_opts = {
+          raise_on_fail: false,
+        }
+
+        cmd = ["clone", nil, "--mirror", remote, local]
+
         if opts[:bare]
-          cmd << "--bare"
+          cmd[1] = "--bare"
         end
```

Comment:
I was trying to cleverly make sure the opts were getting added before the args, although git doesn't really seem to care where they appear. Using the nil value and the numeric index is pretty kludgy though, so I'm switching this to use Array#insert instead. (Even though Array#insert is one of those annoying Ruby methods that both mutates and returns the receiver.)